components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilder.java

```diff
@@ -501,13 +501,7 @@ private void appendParamsToUri(StringBuilder serverUrl, String resolvedParamsStr
                 serverUrl.setLength(serverUrl.length() - 1);
             }
             if (StringUtils.isNotBlank(resolvedParamsString)) {
-                try {
-                    serverUrl.append(delimiter).append(URLEncoder.encode(resolvedParamsString,
-                            StandardCharsets.UTF_8.name()));
-                } catch (UnsupportedEncodingException e) {
-                    throw new URLBuilderException(String.format("Error while trying to build url. %s is not supported" +
-                            ".", StandardCharsets.UTF_8.name()), e);
-                }
+                serverUrl.append(delimiter).append(resolvedParamsString);
```

Comment:
Do we encode the parameters when building resolvedParamsString?

Reply:
do you mean individual parameters?

Reply:
Yes

Reply:
It was not encoded previously. Added fix at https://github.com/wso2/carbon-identity-framework/pull/3011/commits/f4f1ea2d22cbde21781861b96a7a230a312daada